tools/run_tests/xds_k8s_test_driver/framework/test_app/base_runner.py

```diff
@@ -35,14 +50,113 @@ class KubernetesBaseRunner:
     def __init__(self,
                  k8s_namespace,
                  namespace_template=None,
-                 reuse_namespace=False):
+                 reuse_namespace=False,
+                 case_name=None,
+                 pod_label_selector=None):
         # Kubernetes namespaced resources manager
         self.k8s_namespace: k8s.KubernetesNamespace = k8s_namespace
         self.reuse_namespace = reuse_namespace
         self.namespace_template = namespace_template or 'namespace.yaml'
+        self.case_name: str = case_name or ""
+        self.pod_label_selector: str = pod_label_selector or ""
 
         # Mutable state
         self.namespace: Optional[k8s.V1Namespace] = None
+        self._log_stop_event: threading.Event = threading.Event()
+
+    def _await_container_ready(self, core: client.CoreV1Api, pod_name: str,
+                               container_name: str, namespace: str) -> None:
+        while not self._log_stop_event.is_set():
+            try:
+                pod_status = core.read_namespaced_pod_status(
+                    pod_name, namespace).status
+                candidate_containers = [
+                    container for container in pod_status.container_statuses
+                    if container.name == container_name
+                ]
+                if not candidate_containers:
+                    time.sleep(_BACKOFF_SECONDS)
+                    continue
+                container = candidate_containers[0]
+                if not container.ready:
+                    time.sleep(_BACKOFF_SECONDS)
+                else:
+                    break
+            except kubernetes.client.rest.ApiException:
+                time.sleep(_BACKOFF_SECONDS)
+
+    def _log_container(self, core: client.CoreV1Api, pod_name: str,
+                       container_name: str, namespace: str) -> None:
+        query_restarted = False
+        logfile = os.path.join(
+            _TEST_LOG_BASE_DIR,
+            f"{self.case_name}.{pod_name}.{container_name}.sponge_log.log")
```

Comment:
1) Again, i'd recommend against `sponge_log` - this is too specific to kokoro. Won't fly in staging.
2) AFAIK having `sponge_log` with different name variations for the same test would be treated as different attempts at running the same test, like retries.